### DISCUSSION TITLE
Welcome to functions-framework-dart Discussions!

### DISCUSSION THREAD

**Discussion by subfuzion** [GoogleCloudPlatform/functions-framework-dart#53] [General]
## 👋 Hello, World!

This is a place to connect with other members of our community and have discussions without having to open [issues](https://github.com/GoogleCloudPlatform/functions-framework-dart/issues). We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas and brainstorm.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

If you're feeling extroverted, go ahead and comment below with an introduction of yourself and tell us what excites you about serverless Dart! ✏️ 

*Just as a friendly reminder, we follow [Google's Open Source Community Guidelines](https://opensource.google/conduct/). Please help make this a place where everyone feels welcome!*

**Comment by nickmeinhold** [comment]
Hi, thanks for this amazing work - I'm really hyped about using an event-driven microservice architecture in combination with Flutter and using Dart for everything... so pumped! 

I'd love to contribute, currently I'm having trouble getting the presubmit tests to pass because of "mismatched sdk".  I'm on Mac and have been using the dart that comes with Flutter so I tried installing Dart with homebrew and downgrading to 2.10.0 but homebrew doesn't include a 2.10.0 version and I couldn't seem to get any of the methods I found for installing other versions to work.  I'll keep working on it but wanted to say hello and ask if there might be any plans to have the expected version of dart for the tests match the current stable version?  Thanks! 

EDIT: I got Dart 2.10.0 from the [Archive](https://dart.dev/tools/sdk/archive) and set my path so that `dart --version` gives me "Dart SDK version: 2.10.0 (stable)".  Strangely, running `mono_repo presubmit` now gives "skipped, mismatched sdk" for all tests (previously it was just the tests with SDK: 2.10.0).

**Reply by subfuzion** [reply]
Awesome, thanks for the enthusiasm and desire to contribute! Regarding the presubmit checks, what you're seeing is an unfortunately worded status message from the tool, but it's perfectly okay. As part of comprehensive CI checks we run tests against all supported versions; on your system it's just skipping the version that's not installed. You can safely ignore this.

**Reply by nickmeinhold** [reply]
OK great, thanks for that!

**Comment by jtmuller5** [comment]
Hey everyone, I think this project is awesome and I've set up a handful of dart functions so far. 

My question is, what is best practice for deploying multiple functions? Should I deploy one function that routes to other functions based on the input or should I use dartfn to make one "project" for each function I want to deploy?

In my current project, I need to deploy 5 different functions, 4 of which take one type of input and one of which takes another.

Thanks!

**Reply by kevmoo** [reply]
If you strictly want to follow the spec, you deploy multiple functions.

Like if you want to follow/use EventArch

If not, you don't have to!

**Comment by Laebrye** [comment]
Hi everyone!

I'm just trying out the functions framework. On a previous project we used the firebase functions interop package, which really highlighted to me the power of doing both the front and back end in dart. We put all our data models and logic into shared packages and leveraged them. To me, this is way better than something like protobufs or similar because your models can have shared behaviour as well as shared structures.

So I've been exploring using google cloud run to replicate the callable http functions and also using eventarc to replicate the firestore trigger functionality for a new side project of mine. I tried a while back to do the latter and it didn't work, but I don't think the necessary firestore events were emitting to eventarc at that point in time so am ready to give it another go.

If I can't get the eventarc setup working then I'm prepared to have http functions in Cloud Run and node.js middleware for the triggers. I'm aware that the firebase auth on user delete trigger that I need is also not currently covered by eventarc, so I'm going to have to do this for at least one function anyway.

I'm really excited to build something that's truly full stack dart. So I'll be around, loitering in the discussion boards, chipping in, asking questions. I'd be really keen to contribute in any way I can but am fairly new to OS so might be more of a hinderance than a help. If that ever happens just tell me to go away - I won't take offence.

Thanks for the great project!
